### ISSUE TITLE
Bing maps ignore setTilesScaledToDpi(true)

### ISSUE THREAD

**Issue by secaw** [osmdroid/osmdroid#900]
With a `BingMapTileSource` as online tile source, map view does not enlarge the tiles as expected with `setTilesScaledToDpi(true)` initially set. On high res devices too many tiles are therefore being loaded and map names etc are way too small.
When switching back to a different tile sources `setTilesScaledToDpi(true)` works as expected.
When activating a `BingMapTileSource` again and the visible tiles had been cached, then `setTilesScaledToDpi(true)` also kicks in, but as soon as new tiles are loaded, it seems like `setTilesScaledToDpi(true)` is suddenly being ignored again (map switches to high resolution).

Anybody seen this? Is there a known workaround?

Version of osmdroid the issue relates to:
6.0.0

**Comment by spyhunter99** [comment]
that's strange to say the least

**Comment by spyhunter99** [comment]
I just added some options to the sample menu... 
````
mMapView.setTilesScaledToDpi(!mMapView.isTilesScaledToDpi());
mMapView.invalidate();
````

Appears to be working ok...can you get a video or screen shot of what you're seeing?

**Comment by secaw** [comment]
@spyhunter99 Thanks for testing. I did the same with the sample menu now and you're right, it _seems_ to be working. BUT you can reproduce, when starting sample 'Bing Aerial with Labels' -> zoom in closer. This is now not scaled to DPI, although mapview has been initiated with (open menu, it's checked)! Now uncheck the setting: nothing changes on the map, as obviously it was not scaled before, although it should have been. Now recheck the scaling and map enlarges + keeps working as scaled.

I found the cause now:
In `BingMapTileSource.initMetaData()`, `TileSystem.setTileSize(this.getTileSizePixels())` is being called, which of course overwrites the previous call of `TileSystem.setTileSize(size)` in `MapView.updateTileSizeForDensity(ITileSource aTileSource)`.
So what happens is that once `BingMapTileSource` is being initialised (on first online tile load!), a previously set `tileSize` is being recalculated (reset). This also explains, why setting `MapView.isTilesScaledToDpi()` AFTER first bing tile load is working as expected.

The solution here probably is to just delete `TileSystem.setTileSize(this.getTileSizePixels())` from `BingMapTileSource.initMetaData()`. I don't see a reason for this being called, as no other tileSource actually sets an own `tileSize`. Will do a final test and then PR.

**Comment by spyhunter99** [comment]
Note that the sample app starts this process in a background thread in the application class

**Comment by secaw** [comment]
Ok, what's the impact?

**Comment by spyhunter99** [comment]
the impact is that most likely, the init metadata will be finished before the map is loaded

**Comment by secaw** [comment]
@spyhunter99 Do you mention this as a possible explanation, why it seems to be working ok in sample app opposed to other implementations, or as a possible problem with the proposed fix in #902 ? I'm positive, there's no reason for `TileSystem.setTileSize` to be called from `BingMapTileSource`, as `tileSize` is preset with 256 that wouldn't even be changed anyways. The fix is working perfectly with the sample app as far as I can tell.

**Comment by spyhunter99** [comment]
i think the issue with your app is that it's a race condition. When the sample app fires up, it initializes the bing tile source in a background thread. By the time that's finished, the user (you) are at the "pick a sample" activity. By the time you are looking at a map, the background thread is long finished.
